play.pokemonshowdown.com/src/panel-popups.tsx

```diff
@@ -585,6 +586,100 @@ class AvatarsPanel extends PSRoomPanel {
 	}
 }
 
+class BattleForfeitPanel extends PSRoomPanel {
+	static readonly id = 'forfeit';
+	static readonly routes = ['forfeitbattle'];
+	static readonly location = 'semimodal-popup';
```

Comment:
```suggestion
	static readonly location = 'semimodal-popup';
	static readonly noURL = true;
```

You're missing this, that's why the refresh is broken.